### DISCUSSION TITLE
Target collision: Target path xx encountered twice.

### DISCUSSION THREAD

**Discussion by ozgurerdogan** [psy0rz/zfs_autobackup#164] [Q&A]
Where is the issue here:

ssh s4.xxx.com "zfs-autobackup -v s4d1 D3 --keep-source=20 --keep-target=20 --strip-path=2 --clear-mountpoint --clear-refreservatio --ignore-transfer-errors --no-holds --snapshot-format %Y-%m-%d-%H:%M:%S --destroy-missing=10d && zfs-autobackup -v s4d2 D3 --keep-source=20 --keep-target=20 --strip-path=2 --clear-mountpoint --clear-refreservatio --ignore-transfer-errors --no-holds --snapshot-format %Y-%m-%d-%H:%M:%S --destroy-missing=10d";

And I get:
Target collision: Target path D3 encountered twice.
This is proxmox server.

**Comment by ozgurerdogan** [comment]
If I remove --strip-path=2 then it works but to wrong location. Weird thing is, this happens only on two datasets. Rests are fine and on corret location with --strip-path=2.

**Comment by ozgurerdogan** [comment]
I think problem is, I try to make source rpool/data and target D3 look same but that "data" path makes it difficult.  How can I remove from target D3/data to D3 only?

**Comment by psy0rz** [comment]
You cant, you can only stripp stuff from the beginning of the path. Not from the end.

So just strip less and dont worry about the "wrong" location :)

Some more info is here: https://github.com/psy0rz/zfs_autobackup/wiki#the-results

Normally i would say: dont use strip path and so that it preserves the whole path and never will get collissions.

**Comment by ozgurerdogan** [comment]
Ok after some tries, I used --strip-path=2 for rpool/data > D3 (to strip data)
and 
--strip-path=1 for D2 > D3 (to make them same)
And I also used --force option. Not sure if it also helped.
I also did not used autobackup:offsite1=true as it says in wiki. So far so good. But will do more test on other proxmox machines.

**Comment by ozgurerdogan** [comment]
Best part of this tool for me is: "No changes since..." when using backup to local target and remote target. So there are not lots of snapshots.

Because proxmox's pve-sync tool always creates snapshots.
